src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -105,7 +105,7 @@ public function register_routes() {
 				// Excludes invalid directory name characters: `/:<>*?"|`.
 				'([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)',
 				// Matches the template name.
-				'[\/\w-]+'
```

Comment:
@kozer Thinking about this a little more...

Another approach would be to expand this regex to match for `%` (so it matches `small-header-%cf%84%ce%b5%cf%83%cf%84`), and then run `rawurldecode()` inside of the callback before using the data.

There's a slight backwards compatibility concern with running `rawurlencode()` on all paths, because it will change the data passed to existing endpoints.

What do you think about making this change?

Reply:
I think I can try this. Thanks @danielbachhuber !

Reply:
👍

Reply:
@danielbachhuber surprisingly, `rawurldecode` wasn't needed either. Seems that after matching the route, somewhere in the process, it does that, so everything works out of the box. 

It was a very nice proposal, and very elegant! Thanks!

Reply:
@kozer Nice!!! Looks great 😊